backend/api/views.py

```
@@ -73,24 +84,38 @@
+    def get_permissions(self):
+        if self.action == 'retrieve':
+            return [AllowAny()]
+        return super().get_permissions()
 
 class AnalyticsReportViewSet(viewsets.ModelViewSet):
     queryset = AnalyticsReport.objects.all()
     serializer_class = AnalyticsReportSerializer
+    permission_classes = [IsAuthenticated]
     
-    # Removed permission_classes
+    def get_queryset(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `get_queryset`

(ANN201)